docs/en/observability/profiling-upgrade.asciidoc

```diff
@@ -54,10 +56,359 @@ Integrations Server. To do this:
 
 NOTE: When stopping incoming requests, host-agent replicas back off and retry connecting to the {ecloud} endpoint at increasing time intervals.
 
+[discrete]
+[[profiling-migrate-data]]
+=== Migrate existing profiling data
+
+When upgrading to version 8.9, you can use a migration script to migrate existing data if you meet the following requirements:
+
+* The cluster has been upgraded from 8.8 to 8.9.
+* The migration is performed on a Linux workstation with `jq` and `curl` installed.
+
+If you meet these requirements, follow the steps below to run the migration script. If not, we suggest <<profiling-delete-data, deleting your existing profiling data>> and <<profiling-from-scratch, setting up Universal Profiling from scratch>>.
+
+You should also verify that you have a recent snapshot of your data so you can restore it if there's an error during the migration.
+
+
+To run the migration script:
+
+. Save the script below on a Linux workstation, e.g. as `migrate-8.9.sh`
+. Make it executable: `chmod u+x migrate-8.9.sh`
+. Ensure that `jq` and `curl` are installed on the Linux workstation.
+. Export required environment variables:
+  *  `export ES_USER=<superuser name>`
+  *  `export ES_PASSWORD=<superuser password>`
+  *  `export ES_HOST=<elasticsearch endpoint URL>`
+. Run the migration: `./migrate-8.9.sh`
+. Unset environment variables again: `unset ES_USER ES_PASSWORD ES_HOST`
+
+[%collapsible]
+.Migration Script
+====
+[source,bash]
+----
+#!/usr/bin/env bash

```

Comment:
Marking all of this text and copying it resulted in 2 blank lines to be copied before the shebang, which then resulted in this script being executed by `sh` instead of `bash` on my box where it bailed out at `set -o pipefail`. Not sure if we can do much about this though.

Reply:
That's not great and I could reproduce this. 

I played around a bit with CSS styles on the preview page but couldn't find the culprit. The only thing I noticed is that the behavior is identical to the ES docs. E.g. when you copy the snippet in the [index API docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html#create-document-ids-automatically) you'll also notice the two leading and trailing lines.

@mdbirnstiehl is there a possibility to render a "Copy to clipboard" button at the bottom of the script block (similar to the ES docs) to simplify Copy & Paste for users?

Reply:
As a pretty terrible workaround we could add an `if [ $0 == "sh" ]` and then re-run with bash if necessary. But yeah, fixing the CSS (or whatever is causing this) would probably improve things in general (for all other snippets as well). I also recently noticed empty lines to be copied when following instructions for creating an Elastic setup in K8S and it was rather annoying.

Reply:
Thanks for the workaround. I'll keep this in mind and will add this if we don't find another way to address this (i.e. fixing the CSS).

Reply:
I believe the only code block source that adds the copy button is [source,console]. If that is changed, you'll see the copy button as shown below. But that will impact how the script renders in the UI. Not sure if that is a big deal or not in this instance.

![image](https://github.com/elastic/observability-docs/assets/114418652/68e70657-76f6-4d73-a49c-5a6a3c700bd3)

Reply:
Thanks Mike for the clarification. In that case we'll probably go with the workaround that Joel has proposed in https://github.com/elastic/observability-docs/pull/2995#discussion_r1246477490.

Reply:
I played around with this a bit more and it turns out that 

* different OS configurations behave differently. For example, `/usr/bin/sh` is simply symlinked to `/usr/bin/bash`.
* it's not straightforward to determine in which shell you're actually running (see [whatshell.sh](https://www.in-ulm.de/~mascheck/various/whatshell/whatshell.sh.comments.html) for a rather complete implementation). 

I figured the safest option is to ask the user to run the script in bash by changing the instructions from:

```
./migrate-8.9.sh
```

to

```
bash migrate-8.9.sh
```

This ensures that the command is run in bash regardless whether the shebang is detected.